src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,1748 @@
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`OmDetTurboConfig`]):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+OMDET_TURBO_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Padding will be ignored by default should you provide it.
+
+            Pixel values can be obtained using [`AutoImageProcessor`]. See [`DetrImageProcessor.__call__`] for
+            details.
+
+        classes (BatchEmbedding):
+            Tokenized classes. Several classes can be provided for each tasks, thus each element of classes is itself a
+            BatchEmbedding corresponding to classes for a task. Each classes has the attributes `input_ids` and
+            `attention_mask`.
+
+        tasks (BatchEmbedding):
+            Tokenized tasks. Each tasks has the attributes `input_ids` and `attention_mask`.
+"""
+
+
+class OmDetTurboDecoder(OmDetTurboPreTrainedModel):
+    def __init__(self, config: OmDetTurboConfig):
+        super().__init__(config)
+        hidden_dim = config.decoder_hidden_dim
+        self.num_head = config.decoder_num_heads
+        self.num_queries = config.num_queries
+        self.class_distance_type = config.class_distance_type
+        self.learn_init_query = config.learn_init_query
+        self.output_attentions = config.output_attentions
+        self.output_hidden_states = config.output_hidden_states
+        self.use_return_dict = config.use_return_dict
+
+        # backbone feature projection
+        self.channel_projection_layers = nn.ModuleList(
+            nn.Sequential(nn.Conv2d(x, hidden_dim, 1, bias=False), nn.BatchNorm2d(hidden_dim))
+            for x in config.vision_features_channels
+        )
+        self.task_encoder = OmDetTurboTaskEncoder(config)
+        if config.class_dim != hidden_dim:
+            self.task_project = nn.Linear(config.class_dim, hidden_dim)
+
+        # Transformer module
+        self.decoder = OmDetTurboDeformableTransformerDecoder(config)
+
+        # decoder embedding
+        if self.learn_init_query:
+            self.tgt_embed = nn.Embedding(self.num_queries, hidden_dim)
+        self.query_position_head = OmDetTurboMLP(4, 2 * hidden_dim, hidden_dim, num_layers=2)
```

Comment:
Should be created with keyword arguments to make it clearer what these values represent